src/util/message.h

```diff
@@ -101,4 +101,19 @@ uint256 MessageHash(const std::string& message, MessageSignatureFormat format);
 
 std::string SigningResultString(const SigningResult res);
 
+/**
+ * Generate the BIP-322 tx corresponding to the given challenge
+ */
+class BIP322Txs {
+private:
+    template<class T1, class T2>
+    BIP322Txs(const T1& to_spend, const T2& to_sign) : m_to_spend{to_spend}, m_to_sign{to_sign} { }
+
+public:
+    static std::optional<BIP322Txs> Create(const CTxDestination& destination, const std::string& message, MessageVerificationResult& result, std::optional<const std::vector<unsigned char>> = std::nullopt);
```

Comment:
nit: could also name the last parameter here
```suggestion
    static std::optional<BIP322Txs> Create(const CTxDestination& destination, const std::string& message, MessageVerificationResult& result, std::optional<const std::vector<unsigned char>> signature = std::nullopt);
```